### DISCUSSION TITLE
Should this be an npm module?

### DISCUSSION THREAD

**Discussion by amandaghassaei** [amandaghassaei/canvas-capture#2] [General]
Hi everbody!
So I made this repo for my own projects, but now it seems like others may be using it.  I'm wondering if you would like to see this as an npm package so it would be easier to manage versions?  I would hate to push a commit and break someone else's project.

In the meantime you can ref a specific commit when you import this lib using the following syntax:

`npm install github:amandaghassaei/canvas-capture#9307b47`

Let me know if there is interest with a :thumbsup:

**Comment by r4881t** [comment]
I think it should be a separate npm package. It will be easier to install it. For example. I am using `lerna` monorepo and something like `lerna add github:amandaghassaei/canvas-capture` doesn't work out of the box. However, I can install it using `lerna exec yarn add github:amandaghassaei/canvas-capture`.

**Reply by amandaghassaei** [reply]
ok good to know, I'll try to set this up in the next few weeks.
thanks!

**Reply by amandaghassaei** [reply]
@r4881t actually, given the dependency of this repo on git+https://github.com/amandaghassaei/ccapture.js.git#npm-fix it might be a little too messy to make this an official package – I don't want to have to make a copy of CCapture within this repo.  Going to put this on ice for now.

**Reply by amandaghassaei** [reply]
fyi I just put this on npm: `npm install canvas-capture`

**Comment by robertmassman** [comment]
Thanks for letting me know Amanda. I saw yesterday that you were pushing
updates. I can’t wait to see and implement the changes.

Thank you so much for working on this. Hopefully some day I know how to
wrap code in packages so I can contribute to the community.

Happy New Year
Rob



On Fri, Dec 31, 2021 at 11:51 AM Amanda Ghassaei ***@***.***>
wrote:

> fyi I just put this on npm: npm install canvas-capture
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/amandaghassaei/canvas-capture/discussions/2#discussioncomment-1892191>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC72KS2W63PCGAMTRJIPKLLUTX3UVANCNFSM5GF2UZ4Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: <amandaghassaei/canvas-capture/repo-discussions/2/comments/1892191@
> github.com>
>
-- 

Robert Massman LLC
M. 718-813-6870
E. ***@***.***
W. www.robertmassman.com

**Comment by robertmassman** [comment]
Hi Amanda,

Thanks so much for all the work you have put into canvas capture.
I've never used automator before and I'm trying to set up the
conversion script you posted. I'm a little stumped and cannot get it to
convert a webm file on my desktop. Could you please look at the below
screenshot and let me know if you see any errors.

When I right click the file to run the quick action a gear icon appears in
the menu bar but a converted mp4 file is not created.

[image: Screen Shot 2022-03-15 at 6.53.52 AM.png]
Thanks
Rob


-- 

Robert Massman LLC
M. 718-813-6870
E. ***@***.***
W. www.robertmassman.com

**Reply by amandaghassaei** [reply]
Hi Robert,
I'm not able to see the screenshot you posted.  Maybe try running the ffmpeg script from the terminal so you can see any errors printed as it runs:

`ffmpeg -i PATH/FILENAME.webm -vf "crop=trunc(iw/2)*2:trunc(ih/2)*2" -c:v libx264 -preset slow -crf 22 -pix_fmt yuv420p -an PATH/FILENAME.mp4`

does that work for you?
Amanda

**Comment by robertmassman** [comment]
Thanks for the quick reply Amanda and suggestion. I’ll try that when I get
back home.

Rob



On Tue, Mar 15, 2022 at 8:20 AM Amanda Ghassaei ***@***.***>
wrote:

> Hi Robert,
> I'm not able to see the screenshot you posted. Maybe try running the
> ffmpeg script from the terminal so you can see any errors printed as it
> runs:
>
> ffmpeg -i PATH/FILENAME.webm -vf "crop=trunc(iw/2)*2:trunc(ih/2)*2" -c:v
> libx264 -preset slow -crf 22 -pix_fmt yuv420p -an PATH/FILENAME.mp4
>
> does that work for you?
> Amanda
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/amandaghassaei/canvas-capture/discussions/2#discussioncomment-2364636>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC72KS52MCIRLKR7X65CTY3VAB6CFANCNFSM5GF2UZ4Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 

Robert Massman LLC
M. 718-813-6870
E. ***@***.***
W. www.robertmassman.com
